### PR TITLE
Upgrade to kube-java-client 7.x

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-common-ext/kube/src/main/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeClusterMembershipConnectorComponent.java [hunk 1]
```diff
@@ -70,7 +69,7 @@ public ApiClient getApiClient(KubeConnectorConfiguration configuration) {
             client = Config.fromUrl(kubeApiServerUri);
         }
 
-        client.getHttpClient().setReadTimeout(0, TimeUnit.SECONDS); // infinite timeout
+        client.setReadTimeout(0); // infinite timeout
```
Comment (corindwyer) [inline titus-common-ext/kube/src/main/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeClusterMembershipConnectorComponent.java]: can you see if this is still needed?

File: titus-common-ext/kube/src/test/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeExternalResource.java [hunk 2]
```diff
@@ -40,7 +39,7 @@ protected void before() throws Throwable {
                 .standard()
                 .setBasePath(String.format("http://%s:7001", kubeServer))
                 .build();
-        client.getHttpClient().setReadTimeout(0, TimeUnit.SECONDS); // infinite timeout
+        client.setReadTimeout(0); // infinite timeout
```
Comment (corindwyer) [inline titus-common-ext/kube/src/test/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeExternalResource.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 3]
```diff
@@ -248,7 +248,15 @@ public void killTask(String taskId) {
         killTaskCounter.increment();
         try {
             logger.info("deleting pod: {}", taskId);
-            kubeApiFacade.getCoreV1Api().deleteNamespacedPod(taskId, KUBERNETES_NAMESPACE, null, null, null, DELETE_GRACE_PERIOD_SECONDS, null, null);
+            kubeApiFacade.getCoreV1Api().deleteNamespacedPod(taskId,
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: can you move `taskId` to new line to make it easier to read?

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 4]
```diff
@@ -675,7 +683,14 @@ private void reconcileNodesAndPods() {
     private void gcNode(V1Node node) {
         String nodeName = node.getMetadata().getName();
         try {
-            kubeApiFacade.getCoreV1Api().deleteNode(nodeName, null, null, null, 0, null, "Background");
+            kubeApiFacade.getCoreV1Api().deleteNode(nodeName,
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 5]
```diff
@@ -712,7 +727,15 @@ private boolean isNodeReadyForGc(V1Node node) {
     private void gcPod(V1Pod pod) {
         String podName = pod.getMetadata().getName();
         try {
-            kubeApiFacade.getCoreV1Api().deleteNamespacedPod(podName, KUBERNETES_NAMESPACE, null, null, null, 0, null, "Background");
+            kubeApiFacade.getCoreV1Api().deleteNamespacedPod(podName,
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 6]
```diff
@@ -248,7 +248,15 @@ public void killTask(String taskId) {
...
+                    KUBERNETES_NAMESPACE,
+                    null,
+                    null,
+                    DELETE_GRACE_PERIOD_SECONDS,
+                    null,
+                    null,
+                    null
+            );
         } catch (JsonSyntaxException e) {
             // this is probably successful. the generated client has the wrong response type
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: should verify if extra catch is still needed

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 7]
```diff
@@ -675,7 +683,14 @@ private void reconcileNodesAndPods() {
...
+            kubeApiFacade.getCoreV1Api().deleteNode(nodeName,
+                    null,
+                    null,
+                    0,
+                    null,
+                    "Background",
+                    null
+            );
         } catch (JsonSyntaxException e) {
             // this is probably successful. the generated client has the wrong response type
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 8]
```diff
@@ -712,7 +727,15 @@ private boolean isNodeReadyForGc(V1Node node) {
...
+                    KUBERNETES_NAMESPACE,
+                    null,
+                    null,
+                    0,
+                    null,
+                    "Background",
+                    null
+            );
         } catch (JsonSyntaxException e) {
             // this is probably successful. the generated client has the wrong response type
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: same as above

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultDirectKubeApiServerIntegrator.java [hunk 9]
```diff
@@ -143,7 +143,15 @@ public void shutdown() {
             Stopwatch timer = Stopwatch.createStarted();
             try {
                 logger.info("Deleting pod: {}", taskId);
-                kubeApiFacade.getCoreV1Api().deleteNamespacedPod(taskId, KUBERNETES_NAMESPACE, null, null, null, DELETE_GRACE_PERIOD_SECONDS, null, null);
+                kubeApiFacade.getCoreV1Api().deleteNamespacedPod(taskId,
```
Comment (corindwyer) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultDirectKubeApiServerIntegrator.java]: same as above

File: titus-common-ext/kube/src/main/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeClusterMembershipConnectorComponent.java (hunk 1 above)
Comment (tbak) [reply titus-common-ext/kube/src/main/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeClusterMembershipConnectorComponent.java]: I do not know why it was set like that. I am just keeping the original behavior.
Even if not needed, I do not want to remove it now.

File: titus-common-ext/kube/src/test/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeExternalResource.java (hunk 2 above)
Comment (tbak) [reply titus-common-ext/kube/src/test/java/com/netflix/titus/ext/kube/clustermembership/connector/KubeExternalResource.java]: I do not know why it was set like that. I am just keeping the original behavior.
Even if not needed, I do not want to remove it now.

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java (hunk 6 above)
Comment (tbak) [reply titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java]: Still fails with the same exception:
```java
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException:
 Expected a string but was BEGIN_OBJECT at line 1 column 2451 path $.status
```

**Pull request merged**